eureka/registry.go

```diff
@@ -126,3 +130,84 @@ func cfAppName(cfApps []plugin_models.GetAppsModel, cfAppGuid string) (string, e
 
 	return "", errors.New("CF App not found")
 }
+
+// Utility for operating on an application instance in the service registry
+
+type InstanceOperation func(authClient httpclient.AuthenticatedClient, accessToken string, eurekaUrl string, eurekaAppName string, instanceId string) error
+
+func OperateOnApplication(cliConnection plugin.CliConnection, srInstanceName string, cfAppName string, authClient httpclient.AuthenticatedClient, instanceIndex *int,
+	servinceInstanceURL func(cliConnection plugin.CliConnection, serviceInstanceName string, accessToken string, authClient httpclient.AuthenticatedClient) (string, error),
+	operate InstanceOperation) (string, error) {
+	accessToken, err := cfutil.GetToken(cliConnection)
+	if err != nil {
+		return "", err
+	}
+
+	eureka, err := servinceInstanceURL(cliConnection, srInstanceName, accessToken, authClient)
+	if err != nil {
+		return "", fmt.Errorf("Error obtaining service registry URL: %s", err)
+	}
+
+	apps, err := getRegisteredAppsWithCfAppName(cliConnection, authClient, accessToken, eureka, cfAppName)
+	if err != nil {
+		return "", err
+	}
+	statusMessage := ""
+	statusTemplate := "Processed service instance %s with index %s\n"
+	if instanceIndex == nil { //Index is omitted, deregister all instances
+		var err error
+		for _, app := range apps {
+			err = operate(authClient, accessToken, eureka, app.eurekaAppName, app.instanceId)
```

Comment:
Should this break out of the for loop if an error is returned by an invocation of the `operate` function? At present it _looks_ like an error `e0` could occur partway through the loop but if the call to `operate` returns a `nil` error then the `e0` error will not be reported at the end of the loop.

Reply:
@georgeharley Yes, good point. I've fixed this behaviour in the next story up: progress reporting. It was existing misbehaviour, so I don't think this story needs to fix it.